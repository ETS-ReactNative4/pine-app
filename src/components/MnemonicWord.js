import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';

import { withTheme } from '../contexts/theme';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  container: {
    width: '49%',
    padding: 4
  },
  wrapper: {
    borderWidth: 1,
    borderRadius: 30,
    padding: 11,
    paddingLeft: 12
  },
  number: {
    position: 'absolute',
    top: 11,
    left: 5,
    fontSize: 11,
    fontWeight: 'bold',
    width: 20,
    textAlign: 'right'
  },
  word: {
    fontFamily: 'Menlo-Regular',
    fontSize: 11,
    paddingLeft: 17
  }
});

class MnemonicWord extends Component {
  render() {
    const { index, word, theme } = this.props;

    return (
      <View style={styles.container}>
        <View style={[styles.wrapper, theme.mnemonicWordWrapper]}>
          <StyledText style={[styles.number, theme.mnemonicWordNumber]}>
            {index + 1}.
          </StyledText>
          <StyledText style={[styles.word, theme.mnemonicWord]}>
            {word}
          </StyledText>
        </View>
      </View>
    );
  }
}

MnemonicWord.propTypes = {
  index: PropTypes.number.isRequired,
  word: PropTypes.string.isRequired,
  theme: PropTypes.object.isRequired
};

export default withTheme(MnemonicWord);
