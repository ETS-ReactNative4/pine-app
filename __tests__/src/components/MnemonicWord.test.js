import React from 'react';
import renderer from 'react-test-renderer';
import MnemonicWord from '../../../src/components/MnemonicWord';

describe('MnemonicWord', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MnemonicWord index={10} word='bulb' />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
