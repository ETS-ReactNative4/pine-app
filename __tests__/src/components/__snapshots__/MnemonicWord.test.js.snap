// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MnemonicWord renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 4,
      "width": "49%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fbfbfb",
        "borderColor": "#DDDDDF",
        "borderRadius": 30,
        "borderWidth": 1,
        "padding": 7,
        "paddingLeft": 12,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "fontFamily": "Menlo-Regular",
            "fontSize": 11,
          },
        ]
      }
    >
      11
      . 
      bulb
    </Text>
  </View>
</View>
`;
